pytorch_lightning/trainer/distrib_data_parallel.py

```diff
@@ -506,9 +506,10 @@ def ddp_train(self, process_idx, q, model, is_master=False, proc_offset=0):
         model.init_ddp_connection(self.global_rank, self.world_size, self.is_slurm_managing_tasks)
 
         # call setup after the ddp process has connected
-        self.setup('fit')
-        if self.is_function_implemented('setup', model):
-            model.setup('fit')
+        if not self.testing:
+            self.setup('fit')
+            if self.is_function_implemented('setup', model):
```

Comment:
why do we need this check for is_implemented? Can we not just call the the method unconditionally, and inheritance will take care of the rest?

Reply:
You mean why `if self.is_function_implemented('setup', model):` is used or why `if self.is_function_implemented('setup', model):` is used within `if not self.testing:`?

Reply:
The former. I suggest to drop the if statement and just call model.setup(). The if condition, as I see it, will always be True.

Reply:
oh yeah! I got it confused with `is_overridden`. Yeah, it's not required.

Reply:
There are more such places where this is done but not required [here](https://github.com/PyTorchLightning/pytorch-lightning/blob/7b4db3045dcc9e6bb0b66e409b25bb2c7fa378f0/pytorch_lightning/trainer/trainer.py#L942), [here](https://github.com/PyTorchLightning/pytorch-lightning/blob/7b4db3045dcc9e6bb0b66e409b25bb2c7fa378f0/pytorch_lightning/trainer/trainer.py#L1054), [here](https://github.com/PyTorchLightning/pytorch-lightning/blob/7b4db3045dcc9e6bb0b66e409b25bb2c7fa378f0/pytorch_lightning/trainer/trainer.py#L1058) and many more. Should I fix them in this PR?

Reply:
@awaelchli

Reply:
The way you have it now looks good to me.

Reply:
Ok.